app/screens/home/channel_list/categories_list/categories/unreads/unreads.tsx

```diff
@@ -47,7 +47,10 @@ const UnreadCategories = ({onChannelSwitch, onlyUnreads, unreadChannels, unreadT
     const styles = getStyleSheet(theme);
 
     const renderItem = useCallback(({item}: {item: ChannelModel}) => {
+        // TODO: ts error about "Property 'serverUrl' is missing in type"
```

Comment:
Maybe it is working, but I am pretty sure this happens because you are not using the `withServerUrl` as it is expected.

What withServerUrl does is add a context. Where you are using `withServerUrl` in this PR is not about adding a context, but actually using the context that already exist to pass down the `serverUrl` property. You will probably have to create a new HOC to do this. Something probably like this.

```
  type GetProps<C> = C extends React.ComponentType<infer P & { serverUrl?: string }> ? P : never;

const withServerUrl = <C extends React.ComponentType<P>, P = GetProps<C>>(Component: C) {
    return (props: P) => {
        const serverUrl = useServerUrl();
        return <Component {...props} serverUrl={serverUrl} /> ;
    }
}
```
    
    const serverUrl

Reply:
This has turned out to be a bit complicated. I don't fully understand what TS is doing here. But I think withServerUrl is adding the serverUrl to the props as expected. It's just we're not typing it correctly. I was able to fix everything except the typing on the props. I've asked Michel for some help, but if he's busy I'll ask around some more.

Reply:
Added the correct type (thanks @michelengelen!).